azavea/district-builder-dtl-pa
# Fix translations issues

## Overview

This fixes issues with the a translations PR that was recently merged (https://github.com/azavea/district-builder-dtl-pa/pull/47). 

### Checklist


### Notes

Translations were working fine before -- I was just doing it wrong.

The way Django translations work is you run `makemessages` and Django looks through whatever files you tell it to look in (in our case namely Python files and templates; see `makelanguagefiles.py`) for text that needs to be translated. It then makes changes to your `.po` files to reflect what text needs to be translated in the files it looked in. Then you update the `.po` files with translations strings. Then you compile them with `compilemessages` and that is what Django uses.

We do all of this in one straight shot in our `makelanguagefiles` command which runs automatically when the containers start up. If your `.po` files are correct, everything works great. If there are issues, it does not. For me, since I hadn't `rsync-back`d the `.po` files, GNU gettext (what Django uses under the covers) got confused since there were missing translations (when translation keys don't exactly match up it will mark them as `fuzzy` and not display those; this happened with the `msgid` of "plan" which resulted in that not being translated; since it is used all of the place many translations were missing). 

## Testing Instructions

As long as `rsync-auto` is on, just check out the branch and restart the server.
